### PR TITLE
Add basic support for route 53 operations

### PR DESCRIPTION
**Pull request by edhollandAL**

### CODE REVIEW

**Comment by edhollandAL** [comment]
note: we'll probably want to squash these commits into one if we prefer the second API :)

File: src/erlcloud_route53.erl [hunk 1]
```diff
@@ -0,0 +1,269 @@
+%% Amazon Route 53
+
+-module(erlcloud_route53).
+
+-export([describe_zones/0, describe_zones/3]).
```
Comment (motobob) [inline src/erlcloud_route53.erl]: alike other modules this need new/2/3 and config/2/3 to actually set route53 part of aws_config.

File: src/erlcloud_route53.erl [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+-export([describe_resource_sets/1, describe_resource_sets/4]).
+
+-export([describe_delegation_sets/0, describe_delegation_sets/3]).
+
+-include_lib("erlcloud/include/erlcloud.hrl").
+-include_lib("erlcloud/include/erlcloud_aws.hrl").
+-include_lib("erlcloud/include/erlcloud_route53.hrl").
+
+-define(API_VERSION, "2013-04-01").
+-define(DEFAULT_MAX_RECORDS, 20).
```
Comment (motobob) [inline src/erlcloud_route53.erl]: is this from AWS docs or some magic?

File: src/erlcloud_route53.erl [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+%% --------------------------------------------------------------------
+%% @doc Describes resource sets for a specific zone_id using default config
+%% @end
+%% --------------------------------------------------------------------
+-spec(describe_resource_sets(ZoneId :: string()) ->
+             {{paged, string()}, list(aws_route53_delegation_set())} |
+             {{paged, {string(), string()}},list(aws_route53_resourceset())} |
+             {ok, list(aws_route53_resourceset())} |
+             {error, term()}).
+describe_resource_sets(ZoneId) ->
```
Comment (motobob) [inline src/erlcloud_route53.erl:188]: please add arity 2 API like `describe_resource_sets(ZoneId, Cfg)` for all

File: src/erlcloud_route53.erl (hunk 1 above)
Comment (motobob) [inline src/erlcloud_route53.erl]: AWS api docs say 100

File: src/erlcloud_route53.erl [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+%% --------------------------------------------------------------------
+%% @doc Describes all zones using default config
+%% @end
+%% --------------------------------------------------------------------
+-spec(describe_zones() ->
+             {{paged, string()}, list(aws_route53_zone())} |
+             {ok, list(aws_route53_zone())} |
+             {error, term()}).
+describe_zones() ->
```
Comment (motobob) [inline src/erlcloud_route53.erl:119]: limit is optional param so i would encourage not to set it by default. refer erlcloud_ec2 module

File: src/erlcloud_route53.erl (hunk 3 above)
Comment (edhollandAL) [inline src/erlcloud_route53.erl:188]: May want to check previous commit which has a lot of stuff like this. I think it makes it a lot more complex overall though.

File: src/erlcloud_route53.erl [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+extract_name_servers(Set) ->
+        case xmerl_xpath:string("NameServers/NameServer", Set) of
+            [] ->
+                undefined;
+            Xml ->
+                [erlcloud_xml:get_text(X) || X <- Xml]
+        end.
+
+extract_resource_set(Rs) ->
+    #aws_route53_resourceset{
```
Comment (motobob) [inline src/erlcloud_route53.erl]: it's nice that it's records, but we may want to consume prop lists too

File: src/erlcloud_route53.erl [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+    describe_zones([], ?DEFAULT_MAX_RECORDS, erlcloud_aws:default_config()).
+
+%% --------------------------------------------------------------------
+%% @doc Describes all zones using passed config, limit + AWS options
+%% @end
+%% --------------------------------------------------------------------
+-spec(describe_zones(Options   :: list({string(), string()}),
+                     Limit     :: integer(),
+                     AwsConfig :: aws_config()) ->
+    {{paged, string()}, list(aws_route53_zone())} |
```
Comment (motobob) [inline src/erlcloud_route53.erl]: as a consumer of the library i would really like to have pagination logic hidden from me.

Comment (alexturkin) [inline src/erlcloud_route53.erl]: +1. you can see some implementation & interface example in `erlcloud_rds` module

Comment (alexturkin) [inline src/erlcloud_route53.erl]: I'd also consider this discussion: https://github.com/gleber/erlcloud/pull/202#discussion_r38967721

about using flatreverse/1 function for optimizatiokn

**Comment by edhollandAL** [comment]
Added GetHostedZone methods + fixed most review comments. Still record -> proplist conversion required

**Comment by alexturkin** [comment]
+1

**Pull request merged**